plugins/modules/nxos_fc_interfaces.py

```diff
@@ -0,0 +1,491 @@
+#   mode: auto
+#   trunk_mode: on
+#   enabled: False
+#   description: $
+#   analytics: fc-nvme
+# commands:
+# - interface fc18/10
+# - analytics type fc-scsi
+# after:
+# - name: fc18/10
+#   speed: auto max 16000
+#   mode: auto
+#   trunk_mode: on
+#   enabled: False
+#   description: $
+#   analytics: fc-all
+
+# After state:
+# ------------
+#
+# switch# show running-config interface all
+# interface fc18/10
+#     analytics type fc-scsi
+#     analytics type fc-nvme
+#     switchport speed auto max 16000
+#     switchport mode auto
+#     switchport description $
+#     switchport trunk mode on
+#     shutdown
+
+# Using replaced
+
+# Before state:
+# -------------
+#
+# switch# show running-config interface all
+# interface fc18/12
+#     analytics type fc-scsi
+#     analytics type fc-nvme
+#     switchport speed auto max 64000
+#     switchport mode auto
+#     switchport description 1
+#     switchport trunk mode on
+#     no shutdown
+
+- name: Replaces device configuration of listed interfaces with provided configuration
+  cisco.nxos.nxos_fc_interfaces:
+    config:
+    - name: fc18/12
+      speed: auto max 64000
+      mode: auto
+      trunk_mode: on
+      enabled: True
+      description: 1
+      analytics: fc-scsi
+    state: replaced
+
+# Task Output
+# -----------
+#
+# before:
+# - name: fc18/12
+#   speed: auto max 64000
+#   mode: auto
+#   trunk_mode: on
+#   enabled: True
+#   description: 1
+#   analytics: fc-all
+# commands:
+# - interface fc18/12
+# - no analytics type fc-all
+# - analytics type fc-scsi
+# after:
+# - name: fc18/12
+#   speed: auto max 64000
+#   mode: auto
+#   trunk_mode: on
+#   enabled: True
+#   description: 1
+#   analytics: fc-scsi
+
+# After state:
+# ------------
+#
+# switch# show running-config interface all
+# interface fc18/12
+#     analytics type fc-scsi
+#     switchport speed auto max 64000
+#     switchport mode auto
+#     switchport description 1
+#     switchport trunk mode on
+#     no shutdown
+
+
+# Using deleted
+
+# Before state:
+# -------------
+#
+# switch# show running-config interface all
+# interface fc1/2
+#     switchport speed 1000
+#     switchport mode E
+#     no switchport description
+#     switchport trunk mode off
+#     no shutdown
+
+- name: Delete or return interface parameters to default settings
+  cisco.nxos.nxos_fc_interfaces:
+    config:
+    - name: fc1/2
+    state: deleted
+
+# Task Output
+# -----------
+#
+# before:
+# - name: fc1/2
+#   speed: 1000
+#   mode: E
+#   trunk_mode: off
+#   enabled: True
+# commands:
+# - interface fc1/2
+# - no switchport speed 1000
+# - no switchport mode E
+# - switchport trunk mode on
+# - shutdown
+# after:
+# - name: fc1/2
+#   speed: auto
+#   mode: auto
+#   trunk_mode: on
+#   enabled: False
+
+# After state:
+# ------------
+#
+# switch# show running-config interface all
+# interface fc1/2
+#     switchport speed auto
+#     switchport mode auto
+#     no switchport description
```

Comment:
Is this the actual output from the device CLI?

Reply:
yes